tests/application/admin/test_views.py

```diff
@@ -430,9 +432,84 @@ def test_form_has_checkbox_for_each_data_source(self, test_app_client, logged_in
         response = test_app_client.get("/admin/data-sources")
         page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
 
-        # TODO: Update to `admin.merge_data_sources`
-        form = page.find_all("form", action=url_for("admin.data_sources"))[1]
+        form = page.find("form", action=url_for("admin.merge_data_sources_post"))
         assert form
 
         assert find_input_for_label_with_text(form, "Police statistics 2019").get("value") == str(ds1.id)
         assert find_input_for_label_with_text(form, "2011 Census of England and Wales").get("value") == str(ds2.id)
+
+
+class TestMergeDataSourcesView:
+    def test_rdu_user_cannot_see_data_source_merge(self, test_app_client, logged_in_rdu_user):
+
+        response = test_app_client.get("/admin/data-sources/merge")
+
+        assert response.status_code == 403
+
+    def test_admin_user_viewing_two_data_source_to_merge(self, test_app_client, logged_in_admin_user):
+
+        organisation = OrganisationFactory.create(name="Home Office")
+
+        data_source_1 = DataSourceFactory.create(
+            title="2019 police statistics",
+            source_url="https://www.gov.uk/statistics/police/2019",
+            publisher=organisation,
+        )
+        data_source_2 = DataSourceFactory.create(
+            title="Police statistics 2019", source_url="https://statistics.gov.uk/police/2019", publisher=organisation
+        )
+
+        response = test_app_client.get(
+            f"/admin/data-sources/merge?data_sources={data_source_1.id}&data_sources={data_source_2.id}"
+        )
+
+        assert response.status_code == 200
+
+        page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+
+        assert "Merge data sources" == page.find("h1").text
+        assert "Merge data sources" == page.find("title").text
+
+        assert "2019 police statistics" in page.text
+        assert "Police statistics 2019" in page.text
+
+    def test_admin_user_attempting_to_merge_non_existent_ids(self, test_app_client, logged_in_admin_user):
+
+        organisation = OrganisationFactory.create(name="Home Office")
+
+        data_source_1 = DataSourceFactory.create(
+            title="2019 police statistics",
+            source_url="https://www.gov.uk/statistics/police/2019",
+            publisher=organisation,
+        )
+
+        response = test_app_client.get(f"/admin/data-sources/merge?data_sources={data_source_1.id}&data_sources=9999")
+
+        assert response.status_code == 400
+
+
+class TestActuallyMergeDataSourcesView:
```

Comment:
I think some slightly clearer naming conventions I've seen historically include the HTTP method, e.g. `TestGetMergeDataSourcesView` and `TestPostMergeDataSourcesView`. I'm not sure "actually" is the best word to use 😅